arviz/stats/stats.py

```diff
@@ -963,22 +963,24 @@ def _gpinv(probs, kappa, sigma):
     return x
 
 
-def r2_score(y_true, y_pred):
+def r2_score(y_true, y_pred,dist=False):
     """R² for Bayesian regression models. Only valid for linear models.
 
     Parameters
     ----------
-    y_true: array-like of shape = (n_samples) or (n_samples, n_outputs)
+    y_true: array-like of shape = (n_outputs,)
         Ground truth (correct) target values.
-    y_pred: array-like of shape = (n_samples) or (n_samples, n_outputs)
+    y_pred: array-like of shape = (n_posterior_samples, n_outputs)
         Estimated target values.
+    dist(defualt=False): Flag indicating whether to return all posterior samples of the Bayesian R² or only summary stats.
```

Comment:
Probably a better name for the argument would be "summary" or "samples". Also the following wording is more consistent with our doc style

summary: If False return all posterior samples of the Bayesian R² if True only the mean and standard deviation. Defaults to False.